src/sentry/static/sentry/app/components/search/sources/index.jsx

```diff
@@ -47,26 +37,25 @@ class SearchSources extends React.Component {
     });
   }
 
-  render() {
+  renderSources(sources, results, idx) {
+    if (idx >= sources.length) {
+      return this.renderResults(results);
+    }
+    let Source = sources[idx];
     return (
-      <ApiSource {...this.props}>
-        {apiArgs => (
-          <FormSource {...this.props}>
-            {formFieldArgs => (
-              <RouteSource {...this.props}>
-                {routeArgs => (
-                  <CommandSource {...this.props}>
-                    {commandArgs =>
-                      this.renderResults(apiArgs, formFieldArgs, routeArgs, commandArgs)}
-                  </CommandSource>
-                )}
-              </RouteSource>
-            )}
-          </FormSource>
-        )}
-      </ApiSource>
+      <Source {...this.props}>
+        {args => {
+          results[idx] = args;
```

Comment:
The recursive call and mutating `results` feels wrong to me, but I can't think of another solution.